### PR TITLE
feature/pdct 1441 Add family_import_id and import_id to GCF document parser

### PR DESCRIPTION
**Pull request by katybaulch**
# What's changed?

- Add family_import_id and import_id to GCF document parser
- Fixed docstring formatting inconsistency
- Refactored document parser code into smaller functions
- Changed document parser tests to use mock data instead of real GCF data & simplified fixtures

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

### CODE REVIEW

**Comment by katybaulch** [comment]
https://linear.app/climate-policy-radar/issue/PDCT-1441/add-family-import-id-to-gcf-document-mapping

File: gcf_data_mapper/parsers/document.py [hunk 1]
```diff
@@ -59,103 +61,216 @@ def contains_invalid_paths(urls: list[str]) -> bool:
     return False
 
 
-def validate_urls(urls: list[str], doc_id: str) -> None:
-    """
-    Validates a list of URLs for empty, duplicate, and malformed entries.
+def validate_urls(urls: list[str], doc_id: str) -> Optional[bool]:
```
Comment (annaCPR) [inline gcf_data_mapper/parsers/document.py:64]: Why does this need to return an Optional? Can't it just be a bool?

Comment (katybaulch) [reply gcf_data_mapper/parsers/document.py:64]: 🤦🏻 of course - updating!

File: tests/integration_tests/test_commands.py
Comment (annaCPR) [inline tests/integration_tests/test_commands.py:1]: l like this! Might steal it for the bulk import tool

**Review by annaCPR — APPROVED** [review]
lgtm

**Pull request merged**